compiler/rustc_session/src/code_stats.rs

```diff
@@ -44,6 +44,10 @@ pub struct FieldInfo {
     pub offset: u64,
     pub size: u64,
     pub align: u64,
+    /// Name of the type of this field.
+    /// Present only if the creator thought that this would be important for identifying the field,
+    /// typically because the field name is uninformative.
+    pub type_name: Option<Symbol>,
```

Comment:
```suggestion
    pub type_name: Option<Ty<'tcx>>,
```

Reply:
The `Ty` we have when constructing `FieldInfo`. is a `rustc_middle::ty::Ty`. `rustc_session`, the crate containing `FieldInfo`, is a dependency of `rustc_middle`, so that would create a dependency cycle. What do you propose we do about that?

Reply:
Oh lol

We could use a hook (`compiler/rustc_middle/src/hooks/mod.rs`) and pull the `print_type_sizes` + `FieldInfo` out of `rustc_session` and down into `rustc_ty_utils`?

That would have the added benefit of making the layout and the size printing code live together.

Reply:
Well hm. `CodeStats` still lives in `Session`. I wonder if there's anywhere global this can live while also being mutable. Oh well, probably too much work.

Reply:
global mutable state... :scream_cat:

Reply:
It's already global mutable state for the record lol

Fixing this would probably just be uplifting it from Session to GlobalCtxt